### ISSUE TITLE
Solve failure: minimize univariate convex quadratic

### ISSUE THREAD

**Issue by odow** [cvanaret/Uno#27]
I tried to solve `min (x - 1)^2` with the modified options `subproblem primal_dual_interior_point` and `linear_solver MUMPS`:

```julia
julia> using JuMP, Uno_jll

julia> model = Model();

julia> @variable(model, x)
x

julia> @objective(model, Min, (x - 1)^2)
x² - 2 x + 1

julia> write_to_file(model, "problem.nl")


julia> Uno_jll.amplexe() do exe
           run(`$exe problem.nl`)
       end
Problem problem.nl_equalityconstrained_boundrelaxed
1 variables, 0 constraints

Options:
- TR_activity_tolerance = 1e-6
- TR_aggressive_decrease_factor = 4
- TR_decrease_factor = 2
- TR_increase_factor = 2
- TR_min_radius = 1e-7
- TR_radius = 10.
- TR_radius_reset_threshold = 1e-4
- armijo_decrease_fraction = 1e-4
- armijo_tolerance = 1e-9
- barrier_damping_factor = 1e-5
- barrier_default_multiplier = 1
- barrier_initial_parameter = 0.1
- barrier_k_epsilon = 10
- barrier_k_mu = 0.2
- barrier_k_sigma = 1e10
- barrier_push_variable_to_interior_k1 = 1e-2
- barrier_push_variable_to_interior_k2 = 1e-2
- barrier_regularization_exponent = 0.25
- barrier_small_direction_factor = 10.
- barrier_tau_min = 0.99
- barrier_theta_mu = 1.5
- barrier_update_fraction = 10
- constraint_relaxation_strategy = feasibility_restoration
- dual_regularization_fraction = 1e-8
- filter_beta = 0.999
- filter_capacity = 50
- filter_fact = 1.25
- filter_gamma = 0.001
- filter_type = standard
- filter_ubd = 1e2
- globalization_mechanism = TR
- globalization_strategy = fletcher_filter_method
- l1_constraint_violation_coefficient = 1
- least_square_multiplier_max_norm = 1e3
- linear_solver = MUMPS
- logger = INFO
- loose_tolerance = 1e-6
- loose_tolerance_consecutive_iteration_threshold = 15
- max_iterations = 2000
- primal_regularization_decrease_factor = 3.
- primal_regularization_fast_increase_factor = 100.
- primal_regularization_initial_factor = 1e-4
- primal_regularization_lb = 1e-20
- primal_regularization_slow_increase_factor = 8.
- progress_norm = L1
- protect_actual_reduction_against_roundoff = no
- regularization_failure_threshold = 1e40
- residual_norm = INF
- residual_scaling_threshold = 100.
- scale_functions = no
- sparse_format = COO
- statistics_TR_radius_column_order = 10
- statistics_barrier_parameter_column_order = 8
- statistics_complementarity_column_order = 105
- statistics_major_column_order = 1
- statistics_minor_column_order = 2
- statistics_objective_column_order = 100
- statistics_print_header_every_iterations = 15
- statistics_regularization_column_order = 21
- statistics_restoration_phase_column_order = 20
- statistics_stationarity_column_order = 104
- statistics_status_column_order = 200
- statistics_step_norm_column_order = 31
- subproblem = primal_dual_interior_point
- switch_to_optimality_requires_linearized_feasibility = yes
- switching_delta = 0.999
- switching_infeasibility_exponent = 2
- threshold_unsuccessful_attempts = 8
- time_limit = inf
- tolerance = 1e-8
- unbounded_objective_threshold = -1e20
┌───────┬─────────┬────────────────┬─────────────┬───────┬────────────────┬──────────────┬─────────────┬──────────────┬─────────────────┬────────────────────────┐
│ iter  │ TR iter │ barrier param. │ TR radius   │ phase │ regularization │ step norm    │ objective   │ stationarity │ complementarity │ status                 │
├───────┼─────────┼────────────────┼─────────────┼───────┼────────────────┼──────────────┼─────────────┼──────────────┼─────────────────┼────────────────────────┤
│ 0     │ -       │ -              │ 10          │ OPT   │ -              │ -            │ 1           │ 2            │ 0               │ initial point          │
├───────┼─────────┼────────────────┼─────────────┼───────┼────────────────┼──────────────┼─────────────┼──────────────┼─────────────────┼────────────────────────┤
│ -     │ -       │ -              │ -           │ -     │ -              │ -            │ -           │ -            │ -               │ The interior-point subproblem has a trust region. This is not implemented yet.
│
└───────┴─────────┴────────────────┴─────────────┴───────┴────────────────┴──────────────┴─────────────┴──────────────┴─────────────────┴────────────────────────┘

Uno (TR feasibility_restoration fletcher_filter_method primal_dual_interior_point)
Thu Oct 17 13:55:03 2024
────────────────────────────────────────
Status:					Failed with suboptimal point
Objective value:			1
Primal feasibility:			0
┌ Stationarity residual:		2
└ Complementarity residual:		0
┌ Feasibility stationarity residual:	2
└ Feasibility complementarity residual:	0
┌ Infeasibility measure:		0
│ Objective measure:			1
└ Auxiliary measure:			0
CPU time:				0.001001s
Iterations:				1
Objective evaluations:			1
Constraints evaluations:		0
Objective gradient evaluations:		1
Jacobian evaluations:			0
Hessian evaluations:			0
Number of subproblems solved:		0
Process(`/Users/oscar/.julia/artifacts/28075b1a85b0db25e0d15d3795f07627a1d27d43/bin/uno_ampl problem.nl`, ProcessExited(0))

shell> cat problem.nl
g3 1 1 0
 1 0 1 0 0 0
 0 1
 0 0
 0 1 0
 0 0 0 1
 0 0 0 0 0
 0 1
 0 0
 0 0 0 0 0
O0 0
o0
n1
o2
v0
v0
x1
0 0
b
3
G0 1
0 -2
```

**Comment by odow** [comment]
If I change the globalization_mechanism to LS
```
sed -i '' 's/globalization_mechanism TR/globalization_mechanism LS/g' uno.options
```
I get
```Julia
julia> Uno_jll.amplexe() do exe
           run(`$exe problem.nl`)
       end
Problem problem.nl_equalityconstrained_boundrelaxed
1 variables, 0 constraints

Options:
- LS_backtracking_ratio = 0.5
- LS_min_step_length = 1e-12
- LS_scale_duals_with_step_length = yes
- armijo_decrease_fraction = 1e-4
- armijo_tolerance = 1e-9
- barrier_damping_factor = 1e-5
- barrier_default_multiplier = 1
- barrier_initial_parameter = 0.1
- barrier_k_epsilon = 10
- barrier_k_mu = 0.2
- barrier_k_sigma = 1e10
- barrier_push_variable_to_interior_k1 = 1e-2
- barrier_push_variable_to_interior_k2 = 1e-2
- barrier_regularization_exponent = 0.25
- barrier_small_direction_factor = 10.
- barrier_tau_min = 0.99
- barrier_theta_mu = 1.5
- barrier_update_fraction = 10
- constraint_relaxation_strategy = feasibility_restoration
- dual_regularization_fraction = 1e-8
- filter_beta = 0.999
- filter_capacity = 50
- filter_fact = 1.25
- filter_gamma = 0.001
- filter_type = standard
- filter_ubd = 1e2
- globalization_mechanism = LS
- globalization_strategy = fletcher_filter_method
- l1_constraint_violation_coefficient = 1
- least_square_multiplier_max_norm = 1e3
- linear_solver = MUMPS
- logger = INFO
- loose_tolerance = 1e-6
- loose_tolerance_consecutive_iteration_threshold = 15
- max_iterations = 2000
- primal_regularization_decrease_factor = 3.
- primal_regularization_fast_increase_factor = 100.
- primal_regularization_initial_factor = 1e-4
- primal_regularization_lb = 1e-20
- primal_regularization_slow_increase_factor = 8.
- progress_norm = L1
- protect_actual_reduction_against_roundoff = no
- regularization_failure_threshold = 1e40
- residual_norm = INF
- residual_scaling_threshold = 100.
- scale_functions = no
- sparse_format = COO
- statistics_LS_step_length_column_order = 10
- statistics_barrier_parameter_column_order = 8
- statistics_complementarity_column_order = 105
- statistics_major_column_order = 1
- statistics_minor_column_order = 2
- statistics_objective_column_order = 100
- statistics_print_header_every_iterations = 15
- statistics_regularization_column_order = 21
- statistics_restoration_phase_column_order = 20
- statistics_stationarity_column_order = 104
- statistics_status_column_order = 200
- statistics_step_norm_column_order = 31
- subproblem = primal_dual_interior_point
- switch_to_optimality_requires_linearized_feasibility = yes
- switching_delta = 0.999
- switching_infeasibility_exponent = 2
- threshold_unsuccessful_attempts = 8
- time_limit = inf
- tolerance = 1e-8
- unbounded_objective_threshold = -1e20
┌───────┬─────────┬────────────────┬─────────────┬───────┬────────────────┬──────────────┬─────────────┬──────────────┬─────────────────┬────────────────────────┐
│ iter  │ LS iter │ barrier param. │ step length │ phase │ regularization │ step norm    │ objective   │ stationarity │ complementarity │ status                 │
├───────┼─────────┼────────────────┼─────────────┼───────┼────────────────┼──────────────┼─────────────┼──────────────┼─────────────────┼────────────────────────┤
│ 0     │ -       │ -              │ -           │ OPT   │ -              │ -            │ 1           │ 2            │ 0               │ initial point          │
ERROR: failed process: Process(`/Users/oscar/.julia/artifacts/28075b1a85b0db25e0d15d3795f07627a1d27d43/bin/uno_ampl problem.nl`, ProcessSignaled(11)) [0]
```

**Comment by cvanaret** [comment]
Ideally, you'd like to use the command line option `-preset ipopt`, which automatically sets the ingredients accordingly. I'm not sure about the error though.

**Comment by odow** [comment]
This might be my fault because I'm not using libblastrampoline correctly. The AmplNLWriter sets some additional environment variables. Let me try again now that we've got sol files etc.

**Comment by odow** [comment]
Closing because this was my fault.
```Julia
julia> using JuMP, Uno_jll, AmplNLWriter

julia> run(
           `curl https://raw.githubusercontent.com/cvanaret/Uno/refs/heads/main/uno.options -o uno.options`,
       )
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  5983  100  5983    0     0  87985      0 --:--:-- --:--:-- --:--:-- 87985
Process(`curl https://raw.githubusercontent.com/cvanaret/Uno/refs/heads/main/uno.options -o uno.options`, ProcessExited(0))

julia> struct UnoSolverCommand <: AmplNLWriter.AbstractSolverCommand end

julia> function AmplNLWriter.call_solver(
           solver::UnoSolverCommand,
           nl_filename::String,
           options::Vector{String},
           stdin::IO,
           stdout::IO,
       )
           Uno_jll.amplexe() do solver_path
               lbt_default_libs =
                   get(ENV, "LBT_DEFAULT_LIBS", AmplNLWriter._get_blas_loaded_libs())
               cmd = `$solver_path -preset ipopt -linear_solver MUMPS -print_solution yes $nl_filename`
               @show cmd
               if !isempty(lbt_default_libs)
                   cmd = addenv(cmd, "LBT_DEFAULT_LIBS" => lbt_default_libs)
               end
               ret = run(pipeline(cmd; stdin = stdin, stdout = stdout))
               if ret.exitcode != 0
                   error("Nonzero exit code: $(ret.exitcode)")
               end
           end
           return replace(nl_filename, "model.nl" => "model.sol")
       end

julia> model = Model(() -> AmplNLWriter.Optimizer(UnoSolverCommand()))
A JuMP Model
├ solver: AmplNLWriter
├ objective_sense: FEASIBILITY_SENSE
├ num_variables: 0
├ num_constraints: 0
└ Names registered in the model: none

julia> @variable(model, x)
x

julia> @objective(model, Min, (x - 1)^2)
x² - 2 x + 1

julia> optimize!(model)
cmd = `/Users/oscar/.julia/artifacts/28075b1a85b0db25e0d15d3795f07627a1d27d43/bin/uno_ampl -preset ipopt -linear_solver MUMPS -print_solution yes /var/folders/bg/dzq_hhvx1dxgy6gb5510pxj80000gn/T/jl_P3JuqT/model.nl`
Problem /var/folders/bg/dzq_hhvx1dxgy6gb5510pxj80000gn/T/jl_P3JuqT/model.nl_scaled_equalityconstrained_boundrelaxed
1 variables, 0 constraints

Options:
- LS_backtracking_ratio = 0.5
- LS_min_step_length = 5e-7
- LS_scale_duals_with_step_length = yes
- armijo_decrease_fraction = 1e-8
- armijo_tolerance = 1e-9
- barrier_damping_factor = 1e-5
- barrier_default_multiplier = 1
- barrier_initial_parameter = 0.1
- barrier_k_epsilon = 10
- barrier_k_mu = 0.2
- barrier_k_sigma = 1e10
- barrier_push_variable_to_interior_k1 = 1e-2
- barrier_push_variable_to_interior_k2 = 1e-2
- barrier_regularization_exponent = 0.25
- barrier_small_direction_factor = 10.
- barrier_tau_min = 0.99
- barrier_theta_mu = 1.5
- barrier_update_fraction = 10
- constraint_relaxation_strategy = feasibility_restoration
- dual_regularization_fraction = 1e-8
- filter_beta = 0.99999
- filter_capacity = 50
- filter_fact = 1e4
- filter_gamma = 1e-8
- filter_sufficient_infeasibility_decrease_factor = 0.9
- filter_type = standard
- filter_ubd = 1e4
- function_scaling_threshold = 100
- globalization_mechanism = LS
- globalization_strategy = waechter_filter_method
- l1_constraint_violation_coefficient = 1000.
- least_square_multiplier_max_norm = 1e3
- linear_solver = MUMPS
- logger = INFO
- loose_tolerance = 1e-6
- loose_tolerance_consecutive_iteration_threshold = 15
- max_iterations = 2000
- primal_regularization_decrease_factor = 3.
- primal_regularization_fast_increase_factor = 100.
- primal_regularization_initial_factor = 1e-4
- primal_regularization_lb = 1e-20
- primal_regularization_slow_increase_factor = 8.
- progress_norm = L1
- protect_actual_reduction_against_roundoff = yes
- regularization_failure_threshold = 1e40
- residual_norm = INF
- residual_scaling_threshold = 100.
- scale_functions = yes
- sparse_format = COO
- statistics_LS_step_length_column_order = 10
- statistics_barrier_parameter_column_order = 8
- statistics_complementarity_column_order = 105
- statistics_major_column_order = 1
- statistics_minor_column_order = 2
- statistics_objective_column_order = 100
- statistics_print_header_every_iterations = 15
- statistics_regularization_column_order = 21
- statistics_restoration_phase_column_order = 20
- statistics_stationarity_column_order = 104
- statistics_status_column_order = 200
- statistics_step_norm_column_order = 31
- subproblem = primal_dual_interior_point
- switch_to_optimality_requires_linearized_feasibility = no
- switching_delta = 1
- switching_infeasibility_exponent = 2
- threshold_unsuccessful_attempts = 8
- time_limit = inf
- tolerance = 1e-8
- unbounded_objective_threshold = -1e20
┌───────┬─────────┬────────────────┬─────────────┬───────┬────────────────┬──────────────┬─────────────┬──────────────┬─────────────────┬────────────────────────┐
│ iter  │ LS iter │ barrier param. │ step length │ phase │ regularization │ step norm    │ objective   │ stationarity │ complementarity │ status                 │
├───────┼─────────┼────────────────┼─────────────┼───────┼────────────────┼──────────────┼─────────────┼──────────────┼─────────────────┼────────────────────────┤
│ 0     │ -       │ -              │ -           │ OPT   │ -              │ -            │ 1           │ 2            │ 0               │ initial point          │
├───────┼─────────┼────────────────┼─────────────┼───────┼────────────────┼──────────────┼─────────────┼──────────────┼─────────────────┼────────────────────────┤
│ 1     │ 1       │ 0.1            │ 1           │ OPT   │ 0              │ 2            │ 1           │ -            │ -               │ rejected (f-type)      │
├───────┼─────────┼────────────────┼─────────────┼───────┼────────────────┼──────────────┼─────────────┼──────────────┼─────────────────┼────────────────────────┤
│ -     │ 2       │ -              │ 0.5         │ -     │ -              │ 1            │ 0           │ 0            │ 0               │ accepted (f-type)      │
└───────┴─────────┴────────────────┴─────────────┴───────┴────────────────┴──────────────┴─────────────┴──────────────┴─────────────────┴────────────────────────┘

Uno (LS feasibility_restoration waechter_filter_method primal_dual_interior_point)
Fri Oct 18 09:42:29 2024
────────────────────────────────────────
Status:					Converged with feasible KKT point
Objective value:			0
Primal feasibility:			0
┌ Stationarity residual:		0
└ Complementarity residual:		0
┌ Feasibility stationarity residual:	0
└ Feasibility complementarity residual:	0
┌ Infeasibility measure:		0
│ Objective measure:			0
└ Auxiliary measure:			0
Primal solution:			1 
┌ Constraint multipliers:		
│ Lower bound multipliers:		0 
└ Upper bound multipliers:		0 
┌ Constraint feasibility multipliers:	
│ Lower bound feasibility multipliers:	0 
└ Upper bound feasibility multipliers:	0 
Objective multiplier:			1
CPU time:				0.001472s
Iterations:				1
Objective evaluations:			3
Constraints evaluations:		0
Objective gradient evaluations:		3
Jacobian evaluations:			0
Hessian evaluations:			1
Number of subproblems solved:		1
```
